packages/@aws-cdk/lambda-layer-awscli/lib/awscli-layer.ts

```diff
@@ -1,19 +1,82 @@
+/* eslint-disable no-console */
+import * as lambda from '@aws-cdk/aws-lambda';
+import { RemovalPolicy, ResourceEnvironment, Stack } from '@aws-cdk/core';
+import { Construct, Node } from 'constructs';
+import * as childproc from 'child_process';
+import * as os from 'os';
 import * as path from 'path';
-import * as lambda from 'aws-cdk-lib/aws-lambda';
-import { FileSystem } from 'aws-cdk-lib';
-import { Construct } from 'constructs';
 
 /**
  * An AWS Lambda layer that includes the AWS CLI.
  */
-export class AwsCliLayer extends lambda.LayerVersion {
+export class AwsCliLayer implements lambda.ILayerVersion {
+
+  public readonly env: ResourceEnvironment;
+  public readonly layerVersionArn: string;
+  public readonly node: Node;
+  public readonly stack: Stack;
+  public readonly compatibleRuntimes?: lambda.Runtime[];
+
+  private readonly packageName: string = 'lambda-layer-awscli-v1';
+  private readonly npmPackage: any;
+  private readonly layer: lambda.LayerVersion;
+  private readonly wellKnownInstallDir: string;
+
   constructor(scope: Construct, id: string) {
-    super(scope, id, {
-      code: lambda.Code.fromAsset(path.join(__dirname, 'layer.zip'), {
-        // we hash the layer directory (it contains the tools versions and Dockerfile) because hashing the zip is non-deterministic
-        assetHash: FileSystem.fingerprint(path.join(__dirname, '../layer')),
-      }),
-      description: '/opt/awscli/aws',
-    });
+    const installParentDir = os.homedir() ?? os.tmpdir();
+    this.wellKnownInstallDir = path.join(installParentDir, '.awscdk/npm-cache');
+
+    const version = this.requireWrapper(path.join(__dirname, '../package.json')).devDependencies[this.packageName];
+    const pathOfModuleIfAlreadyInstalled = require.resolve(`${this.packageName}`);
+    const versionAlreadyInstalled = this.requireWrapper(path.join(pathOfModuleIfAlreadyInstalled, '../../package.json')).version;
```

Comment:
This will ultimately all go into a reusable function/class later, right? (I suppose it could go into `core`)